### ISSUE TITLE
Preview action fails with "Unexpected token U in JSON at position 0"

### ISSUE THREAD

**Issue by jmoseley** [expo/expo-github-action#221]
### Description of the bug
The preview action will fail with an error, saying "Unexpected token U in JSON at position 0". This is after it has successfully published the update.

### To Reproduce

Here is my github action file:

```
name: Build PR Preview
on: pull_request

jobs:
  update:
    name: EAS Update
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
      contents: read
    steps:
      - name: Check for EXPO_TOKEN
        run: |
          if [ -z "${{ secrets.EXPO_TOKEN }}" ]; then
            echo "You must provide an EXPO_TOKEN secret linked to this project's Expo account in this repo's secrets. Learn more: https://docs.expo.dev/eas-update/github-actions"
            exit 1
          fi

      - name: Checkout repository
        uses: actions/checkout@v3

      - name: Setup Node
        uses: actions/setup-node@v3
        with:
          node-version: 18.x
          cache: yarn

      - name: Setup EAS
        uses: expo/expo-github-action@v8
        with:
          eas-version: latest
          token: ${{ secrets.EXPO_TOKEN }}

      - name: Install dependencies
        run: yarn install

      - name: Generate graphql types
        run: yarn generate
        env:
          HASURA_ADMIN_SECRET: ${{ secrets.PROD_HASURA_ADMIN_SECRET }}
          HASURA_ENDPOINT: ${{ secrets.PROD_HASURA_ENDPOINT }}

      - name: Create preview
        uses: expo/expo-github-action/preview@v8
        with:
          command: eas update --channel "preview-${{ github.event.number }}" --message "${{ github.event.pull_request.title }}"
        env:
          GIT_SHA: ${{ github.sha }}
          RELEASE_CHANNEL: preview

      - name: Extract version from app.json
        uses: sergeysova/jq-action@v2
        id: version
        with:
          cmd: jq -r ".expo.version" app.json

      - name: Upload sourcemaps for android
        run: npx bugsnag-source-maps upload-react-native
          --api-key <redacted>
          --platform android
          --app-version "${{ steps.version.outputs.value }}-$(echo ${{ github.sha }} | cut -c1-8)"
          --source-map "./dist/bundles/$(ls ./dist/bundles | grep "android.*.map")"
          --bundle "./dist/bundles/$(ls ./dist/bundles | grep "android.*.js")"

      - name: Upload sourcemaps for ios
        run: npx bugsnag-source-maps upload-react-native
          --api-key <redacted>
          --platform ios
          --app-version "${{ steps.version.outputs.value }}-$(echo ${{ github.sha }} | cut -c1-8)"
          --source-map "./dist/bundles/$(ls ./dist/bundles | grep "ios.*.map")"
          --bundle "./dist/bundles/$(ls ./dist/bundles | grep "ios.*.js")"
```

Here is log output from a failure:
```
Run expo/expo-github-action/preview@v8
/opt/hostedtoolcache/eas-cli/3.12.0/x64/node_modules/.bin/eas --version
eas-cli/3.12.0 linux-x64 node-v18.16.0
Run eas update --channel "preview-63" --message "feat: <redacted>" --json --non-interactive"
  /opt/hostedtoolcache/eas-cli/3.12.0/x64/node_modules/.bin/eas update --channel preview-63 --message feat: <redacted> --json --non-interactive
- Exporting...

<snip>

[expo-cli] Files successfully saved.
  - Exporting...
  [expo-cli] Processing asset bundle patterns:
  - Exporting...
  [expo-cli] - /home/runner/work/mobile-app/mobile-app/**/*
  - Exporting...
  [expo-cli] Dumping asset map
  - Exporting...
  [expo-cli] Dumping source maps
  - Exporting...
  [expo-cli] Preparing additional debugging files
  - Exporting...
  [expo-cli] Export was successful. Your exported files can be found in dist
  - Exporting...
  ✔ Exported bundle(s)
  - Uploading...
  ✔ Uploaded 2 app bundles
  ✔ Uploading assets skipped - no new assets found
  ✔ Channel: preview-63 pointed at branch: preview-63
  - Publishing...
  ✔ Published!
  [
    {
      "id": "<redacted>",
      "createdAt": "2023-05-15T12:27:05.694Z",
      "group": "<redacted>",
      "branch": "preview-63",
      "message": "feat: <redacted>",
      "runtimeVersion": "<redacted>",
      "platform": "ios",
      "manifestPermalink": "https://u.expo.dev/update/<redacted>",
      "isRollBackToEmbedded": false,
      "gitCommitHash": "<redacted>"
    },
    {
      "id": "<redacted>",
      "createdAt": "2023-05-15T12:27:05.694Z",
      "group": "<redacted>",
      "branch": "preview-63",
      "message": "feat: <redacted>",
      "runtimeVersion": "<redacted>",
      "platform": "android",
      "manifestPermalink": "https://u.expo.dev/update/<redacted>",
      "isRollBackToEmbedded": false,
      "gitCommitHash": "<redacted>"
    }
  ]
Error: Unexpected token U in JSON at position 0
```

#### Expected behavior
This step does not fail.

#### Actual behavior
The github action is marked as fail and does not complete.

### Additional context

I have checked, and the branch does include this new change.

**Comment by MrPumpking** [comment]
I got a similar error with a slightly different message:

`Error: Unexpected token C in JSON at position 1`

**Comment by Robert27** [comment]
Same for me:

`Error: Unexpected token e in JSON at position 0`

Did anyone find a workaround / fix?

**Comment by Noitham** [comment]
Same here, following.

**Comment by varelycode** [comment]
I am getting the same issue. Happened when I upgraded to the latest CLI

**Comment by redkean** [comment]
I got the same issue adding environment variables in .env file in my repo.

**Comment by Yalgie** [comment]
Same issue for me yeah, happened after adding in .env variables dynamically in my CI script

**Comment by nderscore** [comment]
This seems to be caused this issue on Expo: https://github.com/expo/expo/issues/23325

The output of `expo config --json` no longer produces valid JSON because of additional logging to stdout related to environment variable loading.

Until it gets fixed on expo, I'm working around it by stripping any characters that come before the first `{` before parsing the output in `loadProjectConfig`:

```diff
-  return JSON.parse(stdout);
+  return JSON.parse(stdout.replace(/^[^{]+/, ''));
```

In case it's helpful, I've posted this workaround on a public fork here: 
https://github.com/nderscore/expo-github-action/tree/workaround

```diff
      - name: 🚀 Create preview
-        uses: expo/expo-github-action/preview@v8
+        uses: nderscore/expo-github-action/preview@workaround
        with:
          command: eas update --auto
```

**Comment by vecchp** [comment]
Still having the same issue even with https://github.com/expo/expo/issues/23325 merged

**Comment by lsarni** [comment]
> Still having the same issue even with [expo/expo#23325](https://github.com/expo/expo/issues/23325) merged

@vecchp It has been merge but there has been no new release for expo, it should be included on Expo SDK 50 whenever that comes out.

**Comment by byCedric** [comment]
The fix from expo/expo#23325 has now been backported to SDK 49 (`expo@49.0.21`). Upgrade your project to the latest patch version to solve this issue.

- `$ bun|npx|yarn expo install expo@latest`
- `$ bun|npx|yarn expo install --fix`
- `$ bun|npx|yarn expo config --json --type public` should now only output JSON.
